main.cpp

```diff
@@ -45,6 +52,44 @@ int main() {
                 .build();
     });
 
+    server.register_handler<HttpMethod::POST>("/search/{id, ip_address}", [](const HttpRequestContext &context) {
+        std::string player_id = context.path_params.at{"id"};
+        std::string player_address = context.path_params.at{"ip_address"};
+        PlayerInfo player(search_id, client_address);
+
+        if(!search_manager.check_for_active_search(player)){
+            return HttpResponse::new_builder().status(204).build();
+        } else {
+            std::pair<unsigned, PlayerInfo> active_searching_player = search_manager.get_player_info_with_search_id();
+            lobby_manager.create_lobby(player, active_searching_player.second, active_searching_player.first);
+            search_manager.erase_first_active_search();
+        };
+
+        return HttpResponse::new_builder()
+                .status(200)
+                .build();
+    });
+
+    server.register_handler<HttpMethod::GET>("/search/{search_id, player_id}", [](const HttpRequestContext &context) {
+        unsigned search_id = context.path_params.at{"search_id"};
+        std::string player_id = context.path_params.at{"player_id"};
+
+        if(!lobby_manager.check_for_active_lobby_with_id(player_id)){
+            return HttpResponse::new_builder().status(204).build();
+        } else {
+            auto [player1, player2, lobby_id] = lobby_manager.get_lobby_with_id(search_id);
+            game_sessions_manager.start_game_session(player1, player2);
+            lobby_manager.erase_first_active_lobby();
+        }
+
+
+        return HttpResponse::new_builder()
+                .status(200)
+                .body(search_id)
```

Comment:
Why is the search Id returned in the GET request?

Reply:
In the docs it says the method should either return an empty request or the game_id. I guess we dont actually need to return it, is what you are saying, since we already have the logic checking. If this is not correct please elaborate further.